File: frontend/src/app/services/notification.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Notification {
  id?: number;
  titre: string;
  message: string;
  cible: string;
  lu: boolean;
  createdAt: string;
}

@Injectable({
  providedIn: 'root'
})
export class NotificationService {
  private apiUrl = 'http://localhost:8081/api/notifications';

  constructor(private http: HttpClient) {}

  /** POST /api/notifications — Créer une notification (admin) */
  creer(notification: { titre: string; message: string }): Observable<Notification> {
    return this.http.post<Notification>(this.apiUrl, notification);
  }

  /** GET /api/notifications — Récupérer toutes les notifications */
  getAll(): Observable<Notification[]> {
    return this.http.get<Notification[]>(this.apiUrl);
  }

  /** PUT /api/notifications/{id}/lu — Marquer comme lue */
  marquerLu(id: number): Observable<Notification> {
    return this.http.put<Notification>(`${this.apiUrl}/${id}/lu`, {});
  }

  /** GET /api/notifications/count — Nombre de non lues */
  countNonLues(): Observable<number> {
    return this.http.get<number>(`${this.apiUrl}/count`);
  }
}

File: backend/src/main/java/com/edukini/edukini_backend/service/NotificationService.java
package com.edukini.edukini_backend.service;

import com.edukini.edukini_backend.model.Notification;
import com.edukini.edukini_backend.repository.NotificationRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class NotificationService {

    @Autowired
    private NotificationRepository notificationRepository;

    /**
     * Créer une notification (envoyée par l'admin)
     */
    public Notification creer(Notification notification) {
        return notificationRepository.save(notification);
    }

    /**
     * Récupérer toutes les notifications (triées par date décroissante)
     */
    public List<Notification> getAll() {
        return notificationRepository.findByCibleOrderByCreatedAtDesc("ENICARTHAGE");
    }

    /**
     * Marquer une notification comme lue
     */
    public Notification marquerLu(Long id) {
        Optional<Notification> opt = notificationRepository.findById(id);
        if (opt.isPresent()) {
            Notification notification = opt.get();
            notification.setLu(true);
            return notificationRepository.save(notification);
        }
        return null;
    }

    /**
     * Compter les notifications non lues
     */
    public long countNonLues() {
        return notificationRepository.countByCibleAndLuFalse("ENICARTHAGE");
    }
}
